flink-runtime/src/main/java/org/apache/flink/runtime/io/network/api/writer/ResultPartitionWriter.java

```diff
@@ -41,7 +41,7 @@
  * ResultPartitionWriter#fail(Throwable)} still needs to be called afterwards to fully release all
  * resources associated the partition and propagate failure cause to the consumer if possible.
  */
-public interface ResultPartitionWriter extends AutoCloseable, AvailabilityProvider {
+public interface ResultPartitionWriter extends AvailabilityProvider {
```

Comment:
(One small comment as I'm not able to do full review): Why did you remove `AutoCloseable`?

Reply:
It is removed by accident. When I am checking where the close method is called, the IDE gives all places calling AutoCloseable#close, I removed it to reduce the result set size but forgot to add it back. Thanks for pointing it out. I will add it back.